### ISSUE TITLE
BatchNorm after Flatten produces exploded BatchNorm

### ISSUE THREAD

**Issue by gianderiu** [onnx/tensorflow-onnx#714]
If I put a BN after a Flatten a set of operators are used instead of a single BN block.
![image](https://user-images.githubusercontent.com/44895812/67233499-9c132f80-f443-11e9-822f-c60262fe6518.png)
All the black blocks (but the Cast operator that is another problem with Flatten) in the picture implement the batch Normalization of this piece of code:
```
        net = kl.BatchNormalization()(net)
        # -> 7x7x16
        net = kl.Conv2D(16, (1, 1), padding = 'same', activation = 'relu')(net)
        # -> 784
        net = kl.Flatten()(net)
        net = kl.BatchNormalization()(net)
        # -> 2
        self.Output = kl.Dense(self.NumClasses, activation = 'sigmoid', name = 'Output')(net)
        
```

**Comment by jignparm** [comment]
@gianderiu , are you trying to convert a TF model, or a Keras model? 

It's most likely for Keras, since you're using Flatten() and Conv2D(). The repository for the Keras-Onnx convertor is https://github.com/onnx/keras-onnx. Please open an issue there if you're still seeing this problem.

**Comment by neuraghe** [comment]
Thank you very much. Indeed I am using Keras.  
I'll try with the proper convert.

**Comment by gianderiu** [comment]
Thank you very much. Indeed I am using Keras.
I'll try with the proper convert.
